app/src/main/java/com/devshish/internship/presentation/ui/player/PlayerFragment.kt

```diff
@@ -0,0 +1,50 @@
+package com.devshish.internship.presentation.ui.player
+
+import android.os.Bundle
+import android.view.View
+import androidx.fragment.app.Fragment
+import com.bumptech.glide.Glide
+import com.devshish.internship.R
+import com.devshish.internship.databinding.FragmentPlayerBinding
+import com.zhuinden.fragmentviewbindingdelegatekt.viewBinding
+import org.koin.androidx.viewmodel.ext.android.viewModel
+
+class PlayerFragment : Fragment(R.layout.fragment_player) {
+
+    private val binding by viewBinding(FragmentPlayerBinding::bind)
+    private val viewModel: PlayerViewModel by viewModel()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        with(viewModel) {
+            binding.ivPlay.setOnClickListener { onPause() }
+            binding.ivPause.setOnClickListener { onPlay() }
+        }
```

Comment:
I think it is better to make 
with(binding){
    ivPlay.setOnClickListener { viewmodel,onPause() }
    .......
}